{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Goal\" data-toc-modified-id=\"Goal-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Goal</a></span><ul class=\"toc-item\"><li><span><a href=\"#Parsed-nt-database\" data-toc-modified-id=\"Parsed-nt-database-1.1\"><span class=\"toc-item-num\">1.1&nbsp;&nbsp;</span>Parsed nt database</a></span><ul class=\"toc-item\"><li><span><a href=\"#Parsing-nt\" data-toc-modified-id=\"Parsing-nt-1.1.1\"><span class=\"toc-item-num\">1.1.1&nbsp;&nbsp;</span>Parsing nt</a></span></li><li><span><a href=\"#Filter-out-very-low-complexity-seqs\" data-toc-modified-id=\"Filter-out-very-low-complexity-seqs-1.1.2\"><span class=\"toc-item-num\">1.1.2&nbsp;&nbsp;</span>Filter out very low complexity seqs</a></span></li><li><span><a href=\"#Creating-the-kraken2-database\" data-toc-modified-id=\"Creating-the-kraken2-database-1.1.3\"><span class=\"toc-item-num\">1.1.3&nbsp;&nbsp;</span>Creating the kraken2 database</a></span><ul class=\"toc-item\"><li><span><a href=\"#taxonomy\" data-toc-modified-id=\"taxonomy-1.1.3.1\"><span class=\"toc-item-num\">1.1.3.1&nbsp;&nbsp;</span>taxonomy</a></span></li><li><span><a href=\"#adding-to-library-(with-masking)\" data-toc-modified-id=\"adding-to-library-(with-masking)-1.1.3.2\"><span class=\"toc-item-num\">1.1.3.2&nbsp;&nbsp;</span>adding to library (with masking)</a></span></li><li><span><a href=\"#db-build\" data-toc-modified-id=\"db-build-1.1.3.3\"><span class=\"toc-item-num\">1.1.3.3&nbsp;&nbsp;</span>db build</a></span></li></ul></li></ul></li><li><span><a href=\"#Database-of-host-genomes\" data-toc-modified-id=\"Database-of-host-genomes-1.2\"><span class=\"toc-item-num\">1.2&nbsp;&nbsp;</span>Database of host genomes</a></span><ul class=\"toc-item\"><li><span><a href=\"#Non-mammals:-Getting-genus-level-reps\" data-toc-modified-id=\"Non-mammals:-Getting-genus-level-reps-1.2.1\"><span class=\"toc-item-num\">1.2.1&nbsp;&nbsp;</span>Non-mammals: Getting genus-level reps</a></span></li><li><span><a href=\"#Non-mammals:-Getting-genus-level-reps\" data-toc-modified-id=\"Non-mammals:-Getting-genus-level-reps-1.2.2\"><span class=\"toc-item-num\">1.2.2&nbsp;&nbsp;</span>Non-mammals: Getting genus-level reps</a></span></li><li><span><a href=\"#Downloading-genomes\" data-toc-modified-id=\"Downloading-genomes-1.2.3\"><span class=\"toc-item-num\">1.2.3&nbsp;&nbsp;</span>Downloading genomes</a></span></li><li><span><a href=\"#Renaming-sequences\" data-toc-modified-id=\"Renaming-sequences-1.2.4\"><span class=\"toc-item-num\">1.2.4&nbsp;&nbsp;</span>Renaming sequences</a></span></li><li><span><a href=\"#Filtering-out-short-contigs\" data-toc-modified-id=\"Filtering-out-short-contigs-1.2.5\"><span class=\"toc-item-num\">1.2.5&nbsp;&nbsp;</span>Filtering out short contigs</a></span></li><li><span><a href=\"#Database-init\" data-toc-modified-id=\"Database-init-1.2.6\"><span class=\"toc-item-num\">1.2.6&nbsp;&nbsp;</span>Database init</a></span><ul class=\"toc-item\"><li><span><a href=\"#adding-to-library-(with-masking)\" data-toc-modified-id=\"adding-to-library-(with-masking)-1.2.6.1\"><span class=\"toc-item-num\">1.2.6.1&nbsp;&nbsp;</span>adding to library (with masking)</a></span></li><li><span><a href=\"#db-build\" data-toc-modified-id=\"db-build-1.2.6.2\"><span class=\"toc-item-num\">1.2.6.2&nbsp;&nbsp;</span>db build</a></span></li><li><span><a href=\"#move-dbs\" data-toc-modified-id=\"move-dbs-1.2.6.3\"><span class=\"toc-item-num\">1.2.6.3&nbsp;&nbsp;</span>move dbs</a></span></li></ul></li></ul></li><li><span><a href=\"#Database-of-all-vertebrata\" data-toc-modified-id=\"Database-of-all-vertebrata-1.3\"><span class=\"toc-item-num\">1.3&nbsp;&nbsp;</span>Database of all vertebrata</a></span></li><li><span><a href=\"#Parsed-nt-database\" data-toc-modified-id=\"Parsed-nt-database-1.4\"><span class=\"toc-item-num\">1.4&nbsp;&nbsp;</span>Parsed nt database</a></span><ul class=\"toc-item\"><li><span><a href=\"#Parsing-nt\" data-toc-modified-id=\"Parsing-nt-1.4.1\"><span class=\"toc-item-num\">1.4.1&nbsp;&nbsp;</span>Parsing nt</a></span></li><li><span><a href=\"#Filter-out-very-low-complexity-seqs\" data-toc-modified-id=\"Filter-out-very-low-complexity-seqs-1.4.2\"><span class=\"toc-item-num\">1.4.2&nbsp;&nbsp;</span>Filter out very low complexity seqs</a></span></li><li><span><a href=\"#Creating-the-kraken2-database\" data-toc-modified-id=\"Creating-the-kraken2-database-1.4.3\"><span class=\"toc-item-num\">1.4.3&nbsp;&nbsp;</span>Creating the kraken2 database</a></span><ul class=\"toc-item\"><li><span><a href=\"#taxonomy\" data-toc-modified-id=\"taxonomy-1.4.3.1\"><span class=\"toc-item-num\">1.4.3.1&nbsp;&nbsp;</span>taxonomy</a></span></li><li><span><a href=\"#adding-to-library-(with-masking)\" data-toc-modified-id=\"adding-to-library-(with-masking)-1.4.3.2\"><span class=\"toc-item-num\">1.4.3.2&nbsp;&nbsp;</span>adding to library (with masking)</a></span></li><li><span><a href=\"#db-build\" data-toc-modified-id=\"db-build-1.4.3.3\"><span class=\"toc-item-num\">1.4.3.3&nbsp;&nbsp;</span>db build</a></span></li></ul></li></ul></li></ul></li><li><span><a href=\"#Notes\" data-toc-modified-id=\"Notes-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Notes</a></span></li><li><span><a href=\"#sessionInfo\" data-toc-modified-id=\"sessionInfo-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>sessionInfo</a></span></li></ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Goal\n",
    "\n",
    "* Creating custom kraken2 databases for hosts in the dataset:\n",
    "  * NCBI-nt sequence with a host taxonomic classification\n",
    "  * all existing genomes (or similar relatives)\n",
    "* Using a high `--confidence` score and filtering out all reads with a classification"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parsed nt database\n",
    "\n",
    "Database of host sequences parsed from nt\n",
    "\n",
    "### Parsing nt\n",
    "\n",
    "```{bash}\n",
    "\n",
    "# (py3_genome) @ rick:/ebio/abt3_projects/databases_no-backup/humann2/Georg_animal\n",
    "\n",
    "# get all taxIDs for animal hosts (all taxonomic levels)\n",
    "\n",
    "cut -f 7 /ebio/abt3_projects/Georg_animal_feces/data/mapping/unified_metadata_complete_190529.tsv | \\\n",
    "  tail -n +2 | sort -u > unified_metadata_complete_190529_taxIDs.txt\n",
    "\n",
    "# Added 1246547 taxID for Proechimys_semispinosus (previously NA)\n",
    "## number of taxIDs: 212\n",
    "\n",
    "# getting child nodes\n",
    "\n",
    "taxonkit list --indent \"\" --ids 10034,10044,100819,100823,100826,10090,101278,10129,101364,101705,10185,102178,103695,103942,111810,111811,111821,117199,1233216,127946,13125,1320375,137523,155067,156569,171269,176039,181096,181117,184237,184711,184712,196523,201439,210855,211598,218793,218836,225396,230659,27610,27675,283035,283924,29065,29067,29077,302418,30338,30390,30422,30520,30532,323367,34869,34880,34924,36229,36239,36245,36305,37601,37610,38527,38626,40830,41261,419612,447135,45802,465717,46689,47308,47729,48139,48150,48155,48156,48668,48745,51300,51331,52619,52631,53277,54133,54292,54496,54556,55149,56213,57570,57571,57865,58317,58319,59474,59529,596110,60710,61852,62902,64668,65484,662561,670354,68724,69300,69811,71011,722674,72542,73334,75140,75839,77149,7823,7962,8010,8022,8032,8038,80427,8168,8284,8324,8384,8407,8501,867383,8790,88116,8839,8843,8847,8869,8895,8919,89252,8928,8964,8969,9031,9054,9103,9157,9187,9209,9231,9261,9268,9272,9315,9316,9317,9337,9351,9355,9358,9365,9394,9463,94993,9586,9595,9606,9611,9612,9615,9627,9644,9654,9657,9659,9662,9668,9685,9689,9691,9711,9720,9742,9768,9785,9794,9796,9798,9807,9809,9823,9825,9833,9838,9858,9860,9861,9874,98928,9894,9897,9913,9925,9927,9938,9940,9970,99804,9983,9986,9993,9996,1246547 | \\\n",
    "  sort -u | perl -ne 'print if ! /^ *$/' > unified_metadata_complete_190529_taxIDs-all.txt\n",
    "\n",
    "## number of taxIDs: 740\n",
    "\n",
    "# parsing nt seqs based on all host taxIDs (and child taxIDs) \n",
    "\n",
    "blastdbcmd -db nt -entry all -outfmt \"%T %a %s\" | egrep -f <(perl -pe 's/(.+)/^$1 /' unified_metadata_complete_190529_taxIDs-all.txt) | perl -pe 's/(.+) (.+) (.+)/>$2|kraken:taxid|$1\\n$3/' > unified_metadata_complete_190529_nt-parsed.fna\n",
    "\n",
    "## file size: 24G\n",
    "## number of seqs: 4163523\n",
    "\n",
    "```\n",
    "\n",
    "### Filter out very low complexity seqs\n",
    "\n",
    "```\n",
    "# (rna-tools) @ rick:/ebio/abt3_projects/databases_no-backup/humann2/Georg_animal\n",
    "\n",
    "bbduk.sh -Xmx100g threads=24 entropy=0.1 entropywindow=50 entropyk=5 overwrite=t in=unified_metadata_complete_190529_nt-parsed.fna out=/tmp/global/nyoungblut/vert_genomes/metadata-190529_nt-parse_entropyGt0p1.fna\n",
    "\n",
    "## file size: 24G\n",
    "## number of seqs: 4163243\n",
    "```\n",
    "\n",
    "### Creating the kraken2 database\n",
    "\n",
    "#### taxonomy\n",
    "\n",
    "```\n",
    "mkdir -p /tmp/global/nyoungblut/vert_genomes/vert_host_nt/\n",
    "\n",
    "/tmp/global/nyoungblut/vert_genomes/kraken2/scripts/kraken2-build --db /tmp/global/nyoungblut/vert_genomes/vert_host_nt/ --download-taxonomy --use-ftp\n",
    "```\n",
    "\n",
    "> NOTE: I had to use updated master branch to have `--use-ftp` option\n",
    "\n",
    "\n",
    "#### adding to library (with masking)\n",
    "\n",
    "```\n",
    "(Bracken2) @ rick:/tmp/global/nyoungblut/vert_genome\n",
    "\n",
    "screen -L -S Krak2B kraken2-build --threads 24 \\\n",
    "  --db /tmp/global/nyoungblut/vert_genomes/vert_host_nt/ \\\n",
    "  --add-to-library /tmp/global/nyoungblut/vert_genomes/metadata-190529_nt-parse_entropyGt0p1.fna\n",
    "```\n",
    "\n",
    "#### db build\n",
    "\n",
    "```\n",
    "# (Bracken2) @ rick:/tmp/global/nyoungblut/vert_genomes/vert_host_nt/\n",
    "\n",
    "screen -L -S Krak2B kraken2-build --build --threads 24 --db /tmp/global/nyoungblut/vert_genomes/vert_host_nt/\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Database of host genomes\n",
    "\n",
    "Using all genomes from genus-level representatives\n",
    "\n",
    "### Non-mammals: Getting genus-level reps\n",
    "\n",
    "```\n",
    "taxIDs in tmp.txt file:\n",
    "\n",
    "10034,10044,100819,100823,100826,10090,101278,10129,101364,101705,10185,102178,103695,103942,111810,111811,111821,117199,1233216,127946,13125,1320375,137523,155067,156569,171269,176039,181096,181117,184237,184711,184712,196523,201439,210855,211598,218793,218836,225396,230659,27610,27675,283035,283924,29065,29067,29077,302418,30338,30390,30422,30520,30532,323367,34869,34880,34924,36229,36239,36245,36305,37601,37610,38527,38626,40830,41261,419612,447135,45802,465717,46689,47308,47729,48139,48150,48155,48156,48668,48745,51300,51331,52619,52631,53277,54133,54292,54496,54556,55149,56213,57570,57571,57865,58317,58319,59474,59529,596110,60710,61852,62902,64668,65484,662561,670354,68724,69300,69811,71011,722674,72542,73334,75140,75839,77149,7823,7962,8010,8022,8032,8038,80427,8168,8284,8324,8384,8407,8501,867383,8790,88116,8839,8843,8847,8869,8895,8919,89252,8928,8964,8969,9031,9054,9103,9157,9187,9209,9231,9261,9268,9272,9315,9316,9317,9337,9351,9355,9358,9365,9394,9463,94993,9586,9595,9606,9611,9612,9615,9627,9644,9654,9657,9659,9662,9668,9685,9689,9691,9711,9720,9742,9768,9785,9794,9796,9798,9807,9809,9823,9825,9833,9838,9858,9860,9861,9874,98928,9894,9897,9913,9925,9927,9938,9940,9970,99804,9983,9986,9993,9996,1246547\n",
    "\n",
    "\n",
    "(py3_genome) @ rick:/tmp/global/nyoungblut/vert_genomes\n",
    "taxonkit lineage -t tmp.txt | taxonkit reformat -t -f \"{g};{s};{S}\" | cut -f 5 | perl -pe 's/;/\\n/g' | perl -ne 'print if /^.+$/' | sort -u | perl -pe 's/\\n/,/g'\n",
    "\n",
    "ncbi-genome-download -o vert_other -H -r 3 -F fasta -s genbank -t 10001,10033,10034,10043,10044,100819,100823,100826,10088,10090,101278,10128,10129,101364,10162,101675,101705,10184,10185,101898,102177,102178,103695,103942,103951,108820,111810,111811,111821,117199,1233215,1233216,1246547,127946,13124,13125,1320375,137523,155067,156568,156569,1574408,161681,171269,175132,176038,176039,181096,181117,184237,184711,184712,1897773,1960649,196523,201437,201439,210855,211598,218792,218793,218835,218836,225396,230659,233201,27609,27610,27671,27675,283033,283035,283924,29065,29066,29067,29077,302418,30338,30389,30390,30409,30420,30422,30520,30532,30610,323367,325165,339868,34868,34869,34879,34880,34923,34924,36229,36239,36244,36245,36257,36283,36292,36304,36305,36801,37601,37610,38526,38527,38625,38626,39193,392815,39620,40829,40830,41260,41261,419612,42163,43162,44233,447134,447135,45802,465717,46689,47307,47308,47727,47729,48138,48139,48150,48155,48156,48284,48667,48668,48745,50391,51299,51300,51331,51357,52201,52618,52619,52631,53276,53277,54132,54133,54136,54292,54360,54491,54496,54555,54556,55149,56213,56266,56298,57569,57570,57571,57865,58316,58317,58318,58319,59474,59529,60710,61852,62902,64668,651673,65484,662560,662561,670354,68724,69300,69509,69811,71011,720582,722674,72542,73334,75140,75839,77141,77149,7822,7823,7956,7961,7962,8009,8010,8016,8022,8028,8032,8033,8038,80427,8166,8168,8283,8284,8324,8383,8384,8399,8407,8500,8501,8523,8556,8583,867383,86974,86976,8703,8789,8790,88115,88116,8835,8839,8842,8843,8847,8852,8867,8869,8894,8895,8918,8919,89252,8927,8928,8952,8956,8963,8964,8968,8969,9030,9031,9053,9054,9102,9103,9154,9157,9181,9186,9187,9207,9209,9260,9261,9266,9268,9271,9272,9312,9315,9316,9317,9336,9337,9350,9351,9353,9355,9357,9358,9364,9365,9379,9394,9463,9469,94992,94993,9554,9586,9592,9593,9595,9605,9606,9611,9612,9615,9625,9627,9639,9644,9653,9654,9656,9657,9658,9659,9661,9662,9665,9668,9682,9685,9688,9689,9691,9710,9711,9717,9720,9741,9742,9757,9766,9768,9784,9785,9789,9790,9794,9796,9798,9806,9807,9808,9809,9822,9823,9825,9832,9833,9836,9838,9857,9858,9859,9860,9861,9867,9871,9874,98927,98928,9893,9894,9896,9897,9903,9913,9922,9925,9926,9927,9933,9935,9938,9940,9969,9970,9973,9980,99804,9983,9984,9986,9992,9993,9996 --dry-run vertebrate_other | sort\n",
    "\n",
    "# Selected one genome per species\n",
    "## If multiple genomes for a species, prioritized refseq\n",
    "# Created a table of accession --> taxID: vert_host_other_asmbl-taxID.tsv\n",
    "```\n",
    "\n",
    "### Non-mammals: Getting genus-level reps\n",
    "\n",
    "```\n",
    "taxIDs in tmp.txt file:\n",
    "\n",
    "10034,10044,100819,100823,100826,10090,101278,10129,101364,101705,10185,102178,103695,103942,111810,111811,111821,117199,1233216,127946,13125,1320375,137523,155067,156569,171269,176039,181096,181117,184237,184711,184712,196523,201439,210855,211598,218793,218836,225396,230659,27610,27675,283035,283924,29065,29067,29077,302418,30338,30390,30422,30520,30532,323367,34869,34880,34924,36229,36239,36245,36305,37601,37610,38527,38626,40830,41261,419612,447135,45802,465717,46689,47308,47729,48139,48150,48155,48156,48668,48745,51300,51331,52619,52631,53277,54133,54292,54496,54556,55149,56213,57570,57571,57865,58317,58319,59474,59529,596110,60710,61852,62902,64668,65484,662561,670354,68724,69300,69811,71011,722674,72542,73334,75140,75839,77149,7823,7962,8010,8022,8032,8038,80427,8168,8284,8324,8384,8407,8501,867383,8790,88116,8839,8843,8847,8869,8895,8919,89252,8928,8964,8969,9031,9054,9103,9157,9187,9209,9231,9261,9268,9272,9315,9316,9317,9337,9351,9355,9358,9365,9394,9463,94993,9586,9595,9606,9611,9612,9615,9627,9644,9654,9657,9659,9662,9668,9685,9689,9691,9711,9720,9742,9768,9785,9794,9796,9798,9807,9809,9823,9825,9833,9838,9858,9860,9861,9874,98928,9894,9897,9913,9925,9927,9938,9940,9970,99804,9983,9986,9993,9996,1246547\n",
    "\n",
    "taxonkit lineage -t tmp.txt | taxonkit reformat -t -f \"{g};{s};{S}\" | cut -f 5 | perl -pe 's/;/\\n/g' | perl -ne 'print if /^.+$/' | sort -u | perl -pe 's/\\n/,/g'\n",
    "\n",
    "ncbi-genome-download -o vert_mam -H -r 3 -F fasta -s genbank -t 10001,10043,10044,100819,100823,100826,10088,10090,101278,10128,10129,101364,10162,101675,101705,10184,10185,101898,102177,102178,103695,103942,103951,108820,111810,111811,111821,117199,1233215,1233216,1246547,127946,13124,13125,1320375,137523,155067,156568,156569,1574408,161681,171269,175132,176038,176039,181096,181117,184237,184711,184712,1897773,1960649,196523,201437,201439,210855,211598,218792,218793,218835,218836,225396,230659,233201,27609,27610,27671,27675,283033,283035,283924,29065,29066,29067,29077,302418,30338,30389,30390,30409,30420,30422,30520,30532,30610,323367,325165,339868,34868,34869,34879,34880,34923,34924,36229,36239,36244,36245,36257,36283,36292,36304,36305,36801,37601,37610,38526,38527,38625,38626,39193,392815,39620,40829,40830,41260,41261,419612,42163,43162,44233,447134,447135,45802,465717,46689,47307,47308,47727,47729,48138,48139,48150,48155,48156,48284,48667,48668,48745,50391,51299,51300,51331,51357,52201,52618,52619,52631,53276,53277,54132,54133,54136,54292,54360,54491,54496,54555,54556,55149,56213,56266,56298,57569,57570,57571,57865,58316,58317,58318,58319,59474,59529,60710,61852,62902,64668,651673,65484,662560,662561,670354,68724,69300,69509,69811,71011,720582,722674,72542,73334,75140,75839,77141,77149,7822,7823,7956,7961,7962,8009,8010,8016,8022,8028,8032,8033,8038,80427,8166,8168,8283,8284,8324,8383,8384,8399,8407,8500,8501,8523,8556,8583,867383,86974,86976,8703,8789,8790,88115,88116,8835,8839,8842,8843,8847,8852,8867,8869,8894,8895,8918,8919,89252,8927,8928,8952,8956,8963,8964,8968,8969,9030,9031,9053,9054,9102,9103,9154,9157,9181,9186,9187,9207,9209,9260,9261,9266,9268,9271,9272,9312,9315,9316,9317,9336,9337,9350,9351,9353,9355,9357,9358,9364,9365,9379,9394,9463,9469,94992,94993,9554,9586,9592,9593,9595,9605,9606,9611,9612,9615,9625,9627,9639,9644,9653,9654,9656,9657,9658,9659,9661,9662,9665,9668,9682,9685,9688,9689,9691,9710,9711,9717,9720,9741,9742,9757,9766,9768,9784,9785,9789,9790,9794,9796,9798,9806,9807,9808,9809,9822,9823,9825,9832,9833,9836,9838,9857,9858,9859,9860,9861,9867,9871,9874,98927,98928,9893,9894,9896,9897,9903,9913,9922,9925,9926,9927,9933,9935,9938,9940,9969,9970,9973,9980,99804,9983,9984,9986,9992,9993,9996 --dry-run vertebrate_mammalian | sort\n",
    "\n",
    "```\n",
    "\n",
    "### Downloading genomes\n",
    "\n",
    "```\n",
    "(py3_genome) @ rick:/tmp/global/nyoungblut/vert_genomes\n",
    "\n",
    "# non-mammals\n",
    "\n",
    "ncbi-genome-download -o vert_other -H -p 24 -r 3 -F fasta -s refseq -A GCA_004143745.1,GCA_004320145.1,GCA_900302645.1,GCA_901001165.1,GCF_000002315.6,GCF_000146605.2,GCF_000691405.1,GCF_000708925.1,GCF_000951615.1,GCF_001522545.3,GCF_002163495.1,GCF_003342905.1,GCF_003850225.1,GCF_004634155.1,GCF_900067755.1 vertebrate_other\n",
    "\n",
    "ncbi-genome-download -o vert_other -H -p 24 -r 3 -F fasta -s genbank -A GCA_004143745.1,GCA_004320145.1,GCA_900302645.1,GCA_901001165.1,GCF_000002315.6,GCF_000146605.2,GCF_000691405.1,GCF_000708925.1,GCF_000951615.1,GCF_001522545.3,GCF_002163495.1,GCF_003342905.1,GCF_003850225.1,GCF_004634155.1,GCF_900067755.1 vertebrate_other\n",
    "\n",
    "\n",
    "# mammals\n",
    "\n",
    "ncbi-genome-download -o vert_mam -H -p 24 -r 3 -F fasta -s refseq -A GCA_001305905.1,GCF_002263795.1,GCA_004027775.1,GCF_000767585.1,GCF_000311805.1,GCF_000002285.3,GCF_001704415.1,GCA_000751575.1,GCA_004027855.1,GCA_000164785.2,GCF_002863925.1,GCF_000696695.1,GCF_000296755.1,GCF_000181335.3,GCA_004027185.1,GCF_000151905.2,GCA_002995585.1,GCF_000001405.38,GCF_000001905.1,GCA_004027085.1,GCF_000001635.26,GCA_001305785.1,GCA_000004035.1,GCA_000191605.1,GCF_000003625.3,GCF_002742125.1,GCF_001857705.1,GCF_002099425.1,GCA_004348235.1,GCA_003071005.1,GCA_001707965.1,GCA_004026625.1,GCA_002215935.1,GCA_004024825.1,GCF_000003025.6,GCA_000612945.1,GCF_003160815.1 vertebrate_mammalian\n",
    "\n",
    "ncbi-genome-download -o vert_mam -H -p 24 -r 3 -F fasta -s genbank -A GCA_001305905.1,GCF_002263795.1,GCA_004027775.1,GCF_000767585.1,GCF_000311805.1,GCF_000002285.3,GCF_001704415.1,GCA_000751575.1,GCA_004027855.1,GCA_000164785.2,GCF_002863925.1,GCF_000696695.1,GCF_000296755.1,GCF_000181335.3,GCA_004027185.1,GCF_000151905.2,GCA_002995585.1,GCF_000001405.38,GCF_000001905.1,GCA_004027085.1,GCF_000001635.26,GCA_001305785.1,GCA_000004035.1,GCA_000191605.1,GCF_000003625.3,GCF_002742125.1,GCF_001857705.1,GCF_002099425.1,GCA_004348235.1,GCA_003071005.1,GCA_001707965.1,GCA_004026625.1,GCA_002215935.1,GCA_004024825.1,GCF_000003025.6,GCA_000612945.1,GCF_003160815.1 vertebrate_mammalian\n",
    "\n",
    "```\n",
    "\n",
    "### Renaming sequences \n",
    "\n",
    "Using `>SEQID|kraken:taxid|TAXID`\n",
    "\n",
    "Created tables for mapping genome accession to taxID: \n",
    "\n",
    "* non-mammals: vert_host_other_asmbl-taxID.tsv\n",
    "* mammals: vert_host_mam_asmbl-taxID.tsv\n",
    "\n",
    "```\n",
    "(py3_genome) @ rick:/tmp/global/nyoungblut/vert_genomes/\n",
    "\n",
    "/ebio/abt3_projects/Georg_animal_feces/code/rename_genome.py vert_host_other_asmbl-taxID.tsv vert_other.fna\n",
    "\n",
    "/ebio/abt3_projects/Georg_animal_feces/code/rename_genome.py vert_host_mam_asmbl-taxID.tsv vert_mam.fna\n",
    "```\n",
    "\n",
    "### Filtering out short contigs\n",
    "\n",
    "```\n",
    "(py3_genome) @ rick:/tmp/global/nyoungblut/vert_genomes/\n",
    "\n",
    "bioawk -cfastx 'length($seq)>=1000{print \">\"$name\"\\n\"$seq}' vert_other.fna > vert_other_filt.fna\n",
    "\n",
    "bioawk -cfastx 'length($seq)>=1000{print \">\"$name\"\\n\"$seq}' vert_mam.fna > vert_mam_filt.fna\n",
    "```\n",
    "\n",
    "Summary of trimming:\n",
    "\n",
    "```\n",
    "vert_other.fna:      735842\n",
    "vert_other_filt.fna: 344581\n",
    "\n",
    "vert_mam.fna:      15392143\n",
    "vert_mam_filt.fna:  3190285\n",
    "```\n",
    "\n",
    "\n",
    "### Database init\n",
    "\n",
    "```\n",
    "#  taxonomy\n",
    "\n",
    "(Bracken2) @ rick:/tmp/global/nyoungblut/vert_genomes/\n",
    "\n",
    "DBDIR=/tmp/global/nyoungblut/vert_genomes/vert_host_genome/\n",
    "\n",
    "mkdir -p $DBDIR\n",
    "\n",
    "/tmp/global/nyoungblut/vert_genomes/kraken2/scripts/kraken2-build --db $DBDIR --download-taxonomy --use-ftp\n",
    "\n",
    "```\n",
    "\n",
    "NOTE: I had to use updated master branch to have --use-ftp option\n",
    "\n",
    "\n",
    "#### adding to library (with masking)\n",
    "\n",
    "```\n",
    "(Bracken2) @ rick:/tmp/global/nyoungblut/vert_genomes/\n",
    "\n",
    "screen -L -S Krak2B-other kraken2-build --threads 24 --db $DBDIR \\\n",
    "  --add-to-library /tmp/global/nyoungblut/vert_genomes/vert_other_filt.fna\n",
    "  \n",
    "screen -L -S Krak2B-mam kraken2-build --threads 24 --db $DBDIR \\\n",
    "  --add-to-library /tmp/global/nyoungblut/vert_genomes/vert_mam_filt.fna\n",
    "```\n",
    "\n",
    "-- TO HERE for /tmp/global2/ --  \n",
    "\n",
    "#### db build\n",
    " \n",
    "\n",
    "```\n",
    "# (Bracken2) @ rick:/tmp/global/nyoungblut/vert_genomes/\n",
    "\n",
    "screen -L -S Krak2B kraken2-build --build --threads 24 --db $DBDIR\n",
    "```\n",
    "\n",
    "#### move dbs\n",
    "\n",
    "```\n",
    "(base) @ rick:/ebio/abt3_projects/databases_no-backup/kraken2\n",
    "\n",
    "mv /tmp/global/nyoungblut/vert_genomes/vert_host_nt/ vertebrata_nt_db\n",
    "mv /tmp/global/nyoungblut/vert_genomes/vert_host_genome/ vertebrata_genome_db\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Database of all vertebrata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-06-14T07:16:40.621363Z",
     "start_time": "2019-06-14T07:16:40.589Z"
    }
   },
   "source": [
    "## Parsed nt database\n",
    "\n",
    "Database of host sequences parsed from nt\n",
    "\n",
    "### Parsing nt\n",
    "\n",
    "```{bash}\n",
    "\n",
    "# (py3_genome) @ rick:/tmp/global2/nyoungblut/vert_genomes/Kraken2_filt/vertebrata_nt\n",
    "\n",
    "# get all lineage taxIDs for all vertebrata\n",
    "\n",
    "taxonkit list --indent \"\" --ids 7742 | sort -u | perl -ne 'print if ! /^ *$/' > vert_taxIDs.txt\n",
    "\n",
    "# number of taxIDs = 87878\n",
    "\n",
    "# parsing nt seqs based on all host taxIDs (and child taxIDs) \n",
    "\n",
    "blastdbcmd -db nt -entry all -outfmt \"%T %a %s\" | \\\n",
    "  egrep -f <(perl -pe 's/(.+)/^$1 /' vert_taxIDs.txt) | \\\n",
    "  perl -pe 's/(.+) (.+) (.+)/>$2|kraken:taxid|$1\\n$3/' > vert_nt-parsed.fna\n",
    "\n",
    "\n",
    "## file size: 73G\n",
    "## number of seqs: 18061099\n",
    "\n",
    "```\n",
    "\n",
    "### Filter out very low complexity seqs\n",
    "\n",
    "```\n",
    "# (rna-tools) @ rick:/tmp/global2/nyoungblut/vert_genomes/Kraken2_filt/vertebrata_nt\n",
    "\n",
    "bbduk.sh -Xmx100g threads=24 entropy=0.1 entropywindow=50 entropyk=5 overwrite=t in=vert_nt-parsed.fna out=vert_nt-parsed_entropyGt0p1.fna\n",
    "\n",
    "## file size: 74G\n",
    "## number of seqs: 18060312\n",
    "```\n",
    "\n",
    "### Creating the kraken2 database\n",
    "\n",
    "#### taxonomy\n",
    "\n",
    "```\n",
    "(Bracken2) @ rick:/tmp/global2/nyoungblut/vert_genomes/Kraken2_filt/vertebrata_nt\n",
    "\n",
    "mkdir -p vertebrata_nt\n",
    "\n",
    "/tmp/global/nyoungblut/vert_genomes/kraken2/scripts/kraken2-build --db vertebrata_nt --download-taxonomy --use-ftp\n",
    "```\n",
    "\n",
    "> NOTE: I had to use updated master branch to have `--use-ftp` option\n",
    "\n",
    "\n",
    "#### adding to library (with masking)\n",
    "\n",
    "```\n",
    "(Bracken2) @ rick:/tmp/global2/nyoungblut/vert_genomes/Kraken2_filt/vertebrata_nt\n",
    "\n",
    "screen -L -S Krak2B kraken2-build --threads 24 \\\n",
    "  --db vertebrata_nt --add-to-library vert_nt-parsed_entropyGt0p1.fna\n",
    "```\n",
    "\n",
    "#### db build\n",
    "\n",
    "```\n",
    "# (Bracken2) @ rick:/tmp/global2/nyoungblut/vert_genomes/Kraken2_filt/vertebrata_nt\n",
    "\n",
    "screen -L -S Krak2B kraken2-build --build --threads 24 --db vertebrata_nt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes\n",
    "\n",
    "* final location of databases: `/ebio/abt3_projects/databases_no-backup/kraken2/`\n",
    "* databases can be used in `ikraken2` rule in LLMGQC version >= 0.8.4"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sessionInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-06-30T07:25:58.911150Z",
     "start_time": "2019-06-30T07:25:58.343Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.4.1 (2017-06-30)\n",
       "Platform: x86_64-pc-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 18.04.1 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS: /ebio/abt3_projects/software/miniconda3_gt4.4/envs/py3_physeq/lib/R/lib/libRblas.so\n",
       "LAPACK: /ebio/abt3_projects/software/miniconda3_gt4.4/envs/py3_physeq/lib/R/lib/libRlapack.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] compiler_3.4.1  magrittr_1.5    IRdisplay_0.4.4 pbdZMQ_0.3-2   \n",
       " [5] tools_3.4.1     htmltools_0.3.6 base64enc_0.1-3 crayon_1.3.4   \n",
       " [9] Rcpp_0.12.17    uuid_0.1-2      stringi_1.2.3   IRkernel_0.8.12\n",
       "[13] jsonlite_1.5    stringr_1.3.1   digest_0.6.15   repr_0.15.0    \n",
       "[17] evaluate_0.10.1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:py3_physeq]",
   "language": "R",
   "name": "conda-env-py3_physeq-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.1"
  },
  "notify_time": "30",
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "288px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
